src/Parser/Mage/Frost/Modules/Items/ShatteredFragmentsOfSindragosa.js

```diff
@@ -0,0 +1,51 @@
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class ShatteredFragmentsOfSindragosa extends Module {
+
+  static dependencies = {
+		combatants: Combatants,
+	};
+
+  damage = 0;
+  cometStormCasts = 0;
+  legendaryProcs = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasHead(ITEMS.SHATTERED_FRAGMENTS_OF_SINDRAGOSA.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const hasCometStormTalent = this.combatants.selected.hasTalent(SPELLS.COMET_STORM_TALENT.id);
+    if (event.ability.guid !== SPELLS.COMET_STORM_TALENT.id || !hasCometStormTalent) {
+      return;
+    }
+    this.cometStormCasts += 1;
+  }
+
+  on_toPlayer_applybuff(event) {
+    if (event.ability.guid !== SPELLS.RAGE_OF_THE_FROST_WYRM.id) {
+      return;
+    }
```

Comment:
Technically this should be `on_toPlayer_removebuff`, because it's when the player loses the buff that they consume it, but it's unlikely to ever make a difference.

Reply:
The Helm actually uses two separate buffs, 

1. Shattered Fragments of Sindragosa, which stacks and counts the number of Frostbolts
2. Rage of the Frost Wyrm which triggers the Comet Storm.

When i was looking through the Events, I thought that Shattered Fragments stacked to 15, and then applied Rage of the Frost Wyrm which triggered the Comet Storm and then the buff was removed, but it appears that Shattered Fragments actually stacks to 14, removes itself and applies Rage of the Frost Wyrm, then the next Frostbolt removes Frost Wyrm and sends the Comet Storm. 

So long story short, ill change it.